csm/core/services/information.py

```diff
@@ -42,3 +46,77 @@ async def check_compatibility(self, **request_body):
             "reason": reason
         }
         return response
+
+    @Log.trace_method(Log.DEBUG)
+    async def get_topology(self):
+        """
+        Method to fetch the cortx topology
+        :param **request_body: Request body kwargs
+        """
+        plugin_response = self._query_deployment_plugin.get_topology()
```

Comment:
Why are you calling get_topology everytime the API is hit? This is a static information that will not change during the course of deployment. 
This should be fetched only once and should be used everywhere

Reply:
Will discuss with provisioner team and accordingly we can add required code